portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,13 +52,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 //        response.setContentType("application/json;");
 //        response.getWriter().println(json);
 
-    Query query = new Query("Memes").addSort("timestamp", SortDirection.DESCENDING);
-
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Memes").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    response.setContentType("text/html;");
+    if( Boolean.parseBoolean( request.getParameter("aRandomOne") ) == true )
```

Comment:
Looks like to are doing two things that are quite different in a method. Consider extracting them into two separate methods to make your code compact and more readable.

For example:

boolean shouldPickARandomOne = Boolean.parseBoolean(...);
if (shouldPickARandomeOne) {
  printRandomMeme(results);
} else {
  printAllMemes(results);
}